pkg/analyzer/analyzer_test.go

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analyzer_test
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"testing"
+
+	"github.com/dominikbraun/graph"
+	analyzer "github.com/guacsec/guac/pkg/analyzer"
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+)
+
+var diffTestFile = "https://raw.githubusercontent.com/guacsec/guac-test/main/hasSbom-pairs/hasSBOM-syft-spdx-k8s.gcr.io-kube-apiserver.v1.24.1.json"
+
+func readTestFileFromHub(fileUrl string) ([]byte, error){
```

Comment:
hmm how about using https://pkg.go.dev/embed here? with test files - can we get a smaller example to test with?